### PR TITLE
docs(tutorial): bring up-to-date with latest `angular-phonecat`/`angular-seed`

### PR DESCRIPTION
**Pull request by gkalpak**
Includes the following changes:
- Switch all links to `https`.
- Switch from `bower` to `npm` and upgrade AngularJS to 1.7.x.
- Replace `getLocationAbsUrl()` with `getCurrentUrl()`.
- Explain how to upgrade dependencies.
- Mention that Protractor might need upgrading.

##
Related PRs:
- angular/angular-phonecat#430
- angular/angular-seed#444

### CODE REVIEW

File: docs/content/tutorial/index.ngdoc [hunk 1]
```diff
@@ -278,6 +281,45 @@ It is good to run the E2E tests whenever you make changes to the HTML views or w
...
+  <p>
+    If that is the case, you can try upgrading Protractor to newer version. For instructions on how
+    to upgrade dependencies see [Updating dependencies](tutorial/#updating-dependencies).
+  </p>
+</div>
+
+
+### Updating dependencies
+
+In order to avoid surprizes, all dependencies listed in `package.json` are pinned to specific
```
Comment (petebacondarwin) [inline docs/content/tutorial/index.ngdoc]: typo: surprises

File: docs/content/tutorial/step_09.ngdoc [hunk 2]
```diff
@@ -203,10 +187,8 @@ code, we put it into a separate file and used the `.config` suffix.
 ```js
   angular.
     module('phonecatApp').
-    config(['$locationProvider', '$routeProvider',
-      function config($locationProvider, $routeProvider) {
-        $locationProvider.hashPrefix('!');
```
Comment (petebacondarwin) [inline docs/content/tutorial/step_09.ngdoc]: 👍

File: docs/content/tutorial/step_13.ngdoc [hunk 3]
```diff
@@ -21,50 +21,34 @@ In this step, we will change the way our application fetches data.
...
   "name": "angular-phonecat",
-  "description": "A starter project for AngularJS",
-  "version": "0.0.0",
-  "homepage": "https://github.com/angular/angular-phonecat",
-  "license": "MIT",
-  "private": true,
+  ...
   "dependencies": {
-    "angular": "1.5.x",
-    "angular-mocks": "1.5.x",
```
Comment (petebacondarwin) [inline docs/content/tutorial/step_13.ngdoc]: no mocks?

**Review by petebacondarwin — APPROVED** [review]
A few minor comments. Good stuff!

File: docs/content/tutorial/step_13.ngdoc (hunk 3 above)
Comment (gkalpak) [reply docs/content/tutorial/step_13.ngdoc]: With the switch from `bower` to `npm`, `angular-mocks` has been moved to `devDependencies (which are not shown here).

File: docs/content/tutorial/index.ngdoc [hunk 4]
```diff
@@ -99,8 +99,8 @@ The tutorial instructions, from now on, assume you are running all commands from
 
 ### Install Node.js
 
-If you want to run the preconfigured local web server and the test tools then you will also need
-[Node.js v4+][node].
+In order to install dependencies (such as the test tools and AngularJS itself) and run the
+preconfigured local web server, you will also need [Node.js v4+][node].
```
[the remark this answers is not in the document]
Comment (gkalpak) [reply docs/content/tutorial/index.ngdoc]: Also, updated angular/angular-phonecat#430 to test against v6 as well.

**Comment by gkalpak** [comment]
@petebacondarwin, updated. PTAL

**Review by petebacondarwin — APPROVED** [review]

**Pull request closed without merging**